Source/JavaScriptCore/bytecode/BytecodeList.rb

```diff
@@ -1044,6 +1044,55 @@
         operand: :offset,
     }
 
+op :resolve_and_get_from_scope,
+    args: {
+        dst: VirtualRegister,
+        scope: VirtualRegister,
+        resolvedScope: VirtualRegister,
+        var: unsigned,
+        # $begin: :private,
+        resolveType: ResolveType,
+        getPutInfo: GetPutInfo,
+        localScopeDepth: unsigned,
+        offset: unsigned,
+    },
+    metadata: {
+        getPutInfo: GetPutInfo,
+        _0: {
+            watchpointSet: WatchpointSet.*,
+            structure: WriteBarrierBase[Structure],
+        },
+        operand: uintptr_t,
+        profile: ValueProfile,
+        resolveType: ResolveType,
```

Comment:
why do we need this and getPutInfo in metadata?

Reply:
I think they are actively used on runtime to update the profiling info. (JITOperations.cpp, JITPropertyAccess.cpp, LLIntSlowPaths.cpp, and CommonSlowPaths.cpp)

Reply:
We should refactor that code so we don't need these duplicate fields. GetPutInfo has a field in it that is ResolvedType. It's entirely duplicative to have that field stored twice.